components/gearScript/fn_applyLoadout.sqf

```diff
@@ -88,11 +88,12 @@ if ((_loadout isEqualType "") and (isPlayer _unit)) exitWith
 
 if (_loadout isEqualType []) then
```

Comment:
If I understand correctly, here we're assuming that we're dealing with a loadout array (strings are handled further above). But it looks like we're expecting a particular kind of loadout array: a CBA extended loadout (in format [loadout, extraData]). However, I do not see where we're checking if the loadout array matches that format.

Could we conceivably receive a loadout here which uses the default ArmA 3 format (so just the inner loadout array)? Because if so, the sub-array assignment using `_copiedLoadout#0` and `_copiedLoadout#1` will fail.